packages/@atjson/source-ckeditor/test/source-classic-build-test.ts

```diff
@@ -216,4 +216,90 @@ describe("@atjson/source-ckeditor classic build", () => {
       }
     ]);
   });
+  test("update children start/end positions to enforce nesting hierarchy", () => {
+    editor.setData(
+      "<strong><em>Bold and italic</em></strong> <strong><a href='https://www.condenast.com'>bold link</a> just bold</strong>"
```

Comment:
We should have a test that has nested structures that aren't solely `$text` nodes, ie, something like
```
<ol><li>List item 1</li><li>List item 2</li></ol>
```
So we should see
```
$root
  list
    list-item
      $text
    list-item
      $text
```

Reply:
And actually, it's possible more important to have a test that checks what used to be exactly overlapping annotations:
```
<ol><li>List item 1</li></ol>
```